src/Components/Film.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+
+import InfoIcon from "@material-ui/icons/Info";
+import CloseIcon from "@material-ui/icons/Close";
+import AddIcon from "@material-ui/icons/Add";
+import RemoveIcon from "@material-ui/icons/RemoveRedEye";
+
+const Film = ({ film, addToWatchList, removeFromWatchList, clearList }) => {
+  const { l, q, i, s, y, rank, id } = film;
+
+  const [more, setMore] = useState(false);
+
+  const [added, setAdded] = useState(
+    localStorage.getItem("watchList")
+      ? JSON.parse(localStorage.getItem("watchList")).find((e) => e.id === id)
+      : false
+  );
+
+  return (
+    <div className="film column" style={{ background: `url(${i.imageUrl})` }}>
+      <div className="line column">
+        <div className="row" style={{ justifyContent: "space-between" }}>
+          <h4>{l}</h4>
+          <div className="row centered">
+            {!added ? (
+              <span
+                className="icon"
+                onClick={() => {
+                  addToWatchList(film);
+                  setAdded(true);
+                }}>
+                <AddIcon />
+              </span>
+            ) : (
+              <span
+                className="icon"
+                onClick={() => {
+                  removeFromWatchList(film);
+                  !clearList && setAdded(false);
+                  clearList && clearList(film);
+                }}>
+                <RemoveIcon />
+              </span>
+            )}
+            <span className="icon" onClick={() => setMore(!more)}>
+              {more ? <CloseIcon /> : <InfoIcon />}
+            </span>
+          </div>
+        </div>
+        <span>Genre: {q}</span>
+        {more && (
+          <>
+            <span>Actors: {s}</span>
+            <div className="row raiting-wrapper">
+              <span>Rank</span>
+              <div
+                className="rank row centered"
+                style={{
+                  width: `${Math.trunc(+rank) / 80}px`,
```

Comment:
[Magic numbers](https://eslint.org/docs/rules/no-magic-numbers) and computations. Move them to constants and functions with naming